cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -588,3 +590,18 @@ func allPodsAreNew(pods []*apiv1.Pod, currentTime time.Time) bool {
 	found, oldest := getOldestCreateTimeWithGpu(pods)
 	return found && oldest.Add(unschedulablePodWithGpuTimeBuffer).After(currentTime)
 }
+
+func getUpcomingNodeInfos(registry *clusterstate.ClusterStateRegistry, nodeInfos map[string]*schedulernodeinfo.NodeInfo) []*apiv1.Node {
```

Comment:
The tests using mocks are unreadable. Do we have test covering this method anyway? Maybe writing a unit test to just cover this one would be beneficial?

Reply:
Just discovered that the tests with mocks are broken as they are stateful. There is a cache used in scale up. I hacked it by passing a distinct node group id. We need to fix it properly soon.

To answer your question, there are tests that cover this method already.

Reply:
The hack doesn't work in all cases, so, I have changed the test case a little. Will discuss with you tomorrow.